inference-engine/src/mkldnn_plugin/nodes/mkldnn_matrix_nms_node.cpp

```diff
@@ -0,0 +1,390 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_matrix_nms_node.h"
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "base.hpp"
+#include "ie_parallel.hpp"
+#include "ngraph/opsets/opset8.hpp"
+#include "ngraph_ops/nms_static_shape_ie.hpp"
+#include "utils/general_utils.h"
+
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+using MatrixNmsIEInternal = ngraph::op::internal::NmsStaticShapeIE<ngraph::op::v8::MatrixNms>;
+
+using ngNmsSortResultType = ngraph::op::util::NmsBase::SortResultType;
+using ngNmseDcayFunction = ngraph::op::v8::MatrixNms::DecayFunction;
+
+bool MKLDNNMatrixNmsNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto nms = std::dynamic_pointer_cast<const MatrixNmsIEInternal>(op);
+        if (!nms) {
+            errorMessage = "Only internal MatrixNms operation is supported";
+            return false;
+        }
+        const auto& attrs = nms->get_attrs();
+        const auto& sortType = attrs.sort_result_type;
+        if (!one_of(sortType, ngNmsSortResultType::NONE, ngNmsSortResultType::SCORE, ngNmsSortResultType::CLASSID)) {
+            errorMessage = "Doest not support SortResultType " + ngraph::as_string(sortType);
+            return false;
+        }
+        const auto& decayType = attrs.decay_function;
+        if (!one_of(decayType, ngNmseDcayFunction::LINEAR, ngNmseDcayFunction::GAUSSIAN)) {
+            errorMessage = "Does not support DcayFunction " + ngraph::as_string(decayType);
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+MKLDNNMatrixNmsNode::MKLDNNMatrixNmsNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr& cache)
+    : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+
+    errorPrefix = "MatrixNMS layer with name '" + getName() + "' ";
+    const auto matrix_nms = std::dynamic_pointer_cast<const MatrixNmsIEInternal>(op);
+
+    if (getOriginalInputsNumber() != 2)
+        IE_THROW() << errorPrefix << "has incorrect number of input edges: " << getOriginalInputsNumber();
+
+    if (getOriginalOutputsNumber() != 3)
+        IE_THROW() << errorPrefix << "has incorrect number of output edges: " << getOriginalOutputsNumber();
+
+    if (!(inDims[NMS_BOXES][0] == inDims[NMS_SCORES][0] && inDims[NMS_BOXES][1] == inDims[NMS_SCORES][2])) {
+        IE_THROW() << errorPrefix << "has incompatible 'boxes' and 'scores' input dmensions";
+    }
+    const std::vector<Precision> supportedFloatPrecision = {Precision::FP32, Precision::BF16};
```

Comment:
let's align with ```MulticlassNms``` and validate precision into ```initSupportedPrimitiveDescriptors```

Reply:
Done